subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/JvmClassHasher.java

```diff
@@ -112,11 +123,12 @@ public HashCode hashJarFile(FileDetails fileDetails) {
                 visit(zipFile, zipEntry, hasher);
             }
         } catch (Exception e) {
-            throw new UncheckedIOException("Could not calculate the signature for Jar file " + file, e);
+            signature = MALFORMED_JAR;
```

Comment:
And if it never opened it, then it is irrelevant and can be ignored. Makes sense.